config/routes.rb

```diff
@@ -1,6 +1,8 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  namespace :api do
+    namespace :v1 do
+      # Defines the root path route ("/")
+      # root "articles#index"
+    end
```

Comment:
A default route should be set up as a fallback (when no API routes is matched) and as a health endpoint (to check when the API is up or down):

```ruby
root "health#show"
```

Then:

```ruby
class HealthController < ApplicationController
  rescue_from(Exception) { render_down }

  def show
    render_up
  end

  private

  def render_up
    render html: html_status(color: "green")
  end

  def render_down
    render html: html_status(color: "red"), status: :internal_server_error
  end

  def html_status(color:)
    %(<html><body style="background-color: #{color}"></body></html>).html_safe
  end
end
```

Taken from https://github.com/rails/rails/pull/46936/files#diff-dfd63e49a9b67adb3b3ada9ac7e38ad7f29945b3f3db66eab4a663e98ac96ab1 that will come in Rails 7

Reply:
@olivierobert Updated [here](https://github.com/nimblehq/ic-rails-huey-toby/pull/25/commits/24ae8461b7c04689dcf0c1aeeaab86c2b7654113)

> A default route should be set up as a fallback (when no API routes is matched) 

After the update, it will not fallback to root, from what I read the correct way is like this;

`match "*unmatched_route", to: "health#show" , via: :all`

Did I miss something? 🙏

Reply:
You must be definitely doing something wrong as `root` mut work: https://guides.rubyonrails.org/routing.html#using-root

Maybe there is any issue with no routes being defined in the `api/v1` namespace. You might need to add a placeholder routes in there:

```ruby
namespace :api do
    namespace :v1 do
      	get '/', to: 'app#index'
    end
end
```

Reply:
@olivierobert 

`root` is showing correctly, once we navigate to `http://0.0.0.0:3000/`

Do I understand correctly that  `http://0.0.0.0:3000/invalid` should navigate to `root` too? 💭